core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/AbstractFromConfigFactory.java

```diff
@@ -52,7 +59,7 @@ protected EntityDescriptor<Solution_> deduceEntityDescriptor(SolutionDescriptor<
         return entityDescriptor;
     }
 
-    protected EntityDescriptor<Solution_> deduceEntityDescriptor(SolutionDescriptor<Solution_> solutionDescriptor) {
+    protected EntityDescriptor<Solution_> getTheOnlyEntityDescriptor(SolutionDescriptor<Solution_> solutionDescriptor) {
         Collection<EntityDescriptor<Solution_>> entityDescriptors = solutionDescriptor.getGenuineEntityDescriptors();
```

Comment:
`getGenuineEntityDescriptors()` is the expensive part (I think you were referring to this in the comment above). IntelliJ Profiler's Method List doesn't show it at all (which means 0 samples) so I don't think it's on the hot path (in a basic scenario).

On the other hand, it's also being called from other places, on of them being `BestSolutionMutationCountStatistic`, which calls it whenever the best solution changes (if the statistic is enabled). I've filed https://issues.redhat.com/browse/PLANNER-2773 for that.